### PR TITLE
Revert the link support of nav.item and the deprecation of nav.link-to

### LINKED ISSUE #878: nav.item does not support links with nested navs
There is a problem with the API introduced in https://github.com/kaliber5/ember-bootstrap/pull/861: the new link params added to `nav.item` will wrap the `{{yield}}` in a `<a>` link, but that turns out not so well when used together with nested navs:

```hbs
<BsNav as |nav|>
  <nav.item @route="support">
    Support
    <BSNav as |nav2|>
      <nav2.item @route="support.faq">FAQ</nav.item>
    </BsNav>
  </nav.item>
</BsNav>
```

The problem here is that the first link will wrap the nested `<ul>` with all its content, including the FAQ link. 😩

Instead of auto-wrapping we could yield the pre-configured `<LinkTo>` component, but that makes things quite verbose:

```hbs
<BsNav as |nav|>
  <nav.item @route="support" as |item|>
    <item.linkTo>Support</item.linkTo>
    <BSNav as |nav2|>
      <nav2.item @route="support.faq" as |item|><item.linkTo>FAQ</item.linkTo></nav.item>
    </BsNav>
  </nav.item>
</BsNav>
```

For context: all this trouble is needed, as BS (at least BS3) expect the `active` class on the `<li>` rathe than the `<a>` (which would just work out of the box). So we need to pass the link information to the `nav.item`, when we don't want the hacky way of doing it as it was before...

I guess this is blocking the 3.0 release until we have this figured out, which I really wanted to get out asap, in case we need to introduce another breaking change! 😖

Any thoughts? @jelhan @nickschot @lolmaus

### PR DESCRIPTION
**Pull request by simonihmig**
Reverts #783/#861, closes #878.

### CODE REVIEW

**Comment by simonihmig** [comment]
>  I would recommend that we add an alias <nav.linkTo> and use that one in the documentation.

Makes sense, I also thought about this, but wanted to add that in a separate PR...

Another problem here: angle bracket invocation of `<LinkTo>` with `@route` and the other args only works with Ember 3.10+. And the polyfill (which I add the `<LinkTo>` support implementation for the now reversed PR! 🙃) does not change that, as it only works for `<LinkTo>`, not `<nav.linkTo>`. 

So we can either support that only when Ember does it natively (users can still use the curly + positional params version), and skip the (angle brackets `nav.link-to`) tests for Ember <3.10, or implement some kind of internal polyfill (which I am not super interested in investing the time for)

**Comment by jelhan** [comment]
> Another problem here: angle bracket invocation of `<LinkTo>` with `@route` and the other args only works with Ember 3.10+. And the polyfill (which I add the `<LinkTo>` support implementation for the now reversed PR! ) does not change that, as it only works for `<LinkTo>`, not `<nav.linkTo>`.
> 
> So we can either support that only when Ember does it natively (users can still use the curly + positional params version), and skip the (angle brackets `nav.link-to`) tests for Ember <3.10, or implement some kind of internal polyfill (which I am not super interested in investing the time for)

Wouldn't this one be resolved if we refactor `<nav.linkTo>` to be a tagless component that uses `<LinkTo>` instead of extending it?

**Comment by simonihmig** [comment]
> Wouldn't this one be resolved if we refactor <nav.linkTo> to be a tagless component that uses <LinkTo> instead of extending it?

Yes, but not sure how we would implement that. The current solution relies on the nav having access to the child links, and especially their `active` property, through the parent/child mixins. If we just use the `LinkComponent` instead of extending it, we make it a "black box" again, and loose access to it, right?

**Comment by jelhan** [comment]
> Yes, but not sure how we would implement that. The current solution relies on the nav having access to the child links, and especially their `active` property, through the parent/child mixins. If we just use the `LinkComponent` instead of extending it, we make it a "black box" again, and loose access to it, right?

Couldn't we replace that one with `isActive` method of `RouterService`? Something like:

```js
export default Component.extend({
  router: service(),

  didReceiveAttrs() {
    let params = [
     this.route,
     Array.isArray(this.models) ? ...this.models : this.model,
     this.queryParams;
   ];
   let isActive = this.get('router').isActive(params);
   // notify parent component about active state
  },
});

**Comment by simonihmig** [comment]
👆That example would be the (tagless) component that *uses* `<LinkTo>`?
If so, it would need to also support positional params, if this shouldn't be a breaking change. And as such it would need to make those also work with the `active` logic, and properly pass those down to `<LinkTo>`. So it's a bit more effort... 

Tbh, I am just a bit tired of investing so much time into that (legacy BS3) active class issue... 🤷‍♂️

File: addon/components/base/bs-nav.js [hunk 1]
```diff
@@ -15,11 +15,15 @@ import layout from 'ember-bootstrap/templates/components/bs-nav';
...
-      Foo
+    <nav.item>
+      <nav.link-to @route="foo">
+        Foo
+      </nav.link-to>
     </nav.item>
-    <nav.item @route="bar" @model={{this.model}}>
-      Bar
+    <nav.item>
+      <nav.link.to @route="bar" @model={{this.model}}>
```
[the remark this answers is not in the document]
Comment (simonihmig) [reply addon/components/base/bs-nav.js]: Fixed

File: tests/dummy/app/templates/demo/navs.hbs [hunk 2]
```diff
@@ -61,10 +65,10 @@
...
-      type=type
-      stacked=stacked
-      justified=justified
-      fill=fill
-    }}
+    type=type
+    stacked=stacked
+    justified=justified
+    fill=fill
+  }}
```
[the remark this answers is not in the document]
Comment (simonihmig) [reply tests/dummy/app/templates/demo/navs.hbs]: Fixed

File: tests/dummy/app/templates/demo/navbars.hbs [hunk 3]
```diff
@@ -60,9 +64,9 @@
...
     @name="navbar-demo.hbs"
     @dynamicProperties={{hash
-      type=type
-      bg=bg
-      collapsed=collapsed
-    }}
+    type=type
+    bg=bg
+    collapsed=collapsed
+  }}
```
[the remark this answers is not in the document]
Comment (simonihmig) [reply tests/dummy/app/templates/demo/navbars.hbs]: Fixed

**Comment by simonihmig** [comment]
Skipping angle bracket `<nav.link-to>` tests now for Ember <3.10.

**Pull request merged**